app/src/main/java/org/simple/clinic/appupdate/CheckAppUpdateAvailability.kt

```diff
@@ -11,35 +12,64 @@ import io.reactivex.Observable
 import org.simple.clinic.appupdate.AppUpdateState.AppUpdateStateError
 import org.simple.clinic.appupdate.AppUpdateState.DontShowAppUpdate
 import org.simple.clinic.appupdate.AppUpdateState.ShowAppUpdate
+import org.simple.clinic.appupdate.CheckAppUpdateAvailability.AppUpdateInfoTaskResult.Failure
+import org.simple.clinic.appupdate.CheckAppUpdateAvailability.AppUpdateInfoTaskResult.Success
 import javax.inject.Inject
 
 class CheckAppUpdateAvailability @Inject constructor(
     private val appContext: Application,
-    private val config: AppUpdateConfig,
+    private val config: Observable<AppUpdateConfig>,
     private val versionUpdateCheck: (Int, Application, AppUpdateConfig) -> Boolean = isVersionApplicableForUpdate
 ) {
 
   fun listen(): Observable<AppUpdateState> {
+    return appUpdateCallback()
+        .flatMap {
+          when (it) {
+            is Success -> shouldNudgeForUpdate(it.appUpdateInfo)
+            is Failure -> Observable.just(AppUpdateStateError(it.exception))
+          }
+        }
+  }
+
+  private fun appUpdateCallback(): Observable<AppUpdateInfoTaskResult> {
     val appUpdateManager = AppUpdateManagerFactory.create(appContext)
     val appUpdateInfoTask = appUpdateManager.appUpdateInfo
 
-    return Observable.create { emitter ->
+    return Observable.create<AppUpdateInfoTaskResult> { emitter ->
       appUpdateInfoTask.addOnSuccessListener { appUpdateInfo ->
-        if (shouldNudgeForAppUpdate(appUpdateInfo)) {
-          emitter.onNext(ShowAppUpdate)
-        } else {
-          emitter.onNext(DontShowAppUpdate)
-        }
+        emitter.onNext(Success(appUpdateInfo))
       }
 
       appUpdateInfoTask.addOnFailureListener { exception ->
-        emitter.onNext(AppUpdateStateError(exception))
+        emitter.onNext(Failure(exception))
       }
     }
   }
 
-  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-  fun shouldNudgeForAppUpdate(appUpdateInfo: AppUpdateInfo): Boolean {
+  private sealed class AppUpdateInfoTaskResult {
+    data class Success(val appUpdateInfo: AppUpdateInfo) : AppUpdateInfoTaskResult()
+    data class Failure(val exception: Exception) : AppUpdateInfoTaskResult()
+  }
+
+  @VisibleForTesting(otherwise = PRIVATE)
+  fun shouldNudgeForUpdate(appUpdateInfo: AppUpdateInfo): Observable<AppUpdateState> {
+    val checkUpdate = config
+        .map { checkForUpdate(appUpdateInfo, it) }
+
+    val shouldShow = checkUpdate
+        .filter { showUpdate -> showUpdate }
+        .map { ShowAppUpdate }
+
+    val doNotShow = config
```

Comment:
We can replace 
```kotlin
config
  .map { checkForUpdate(appUpdateInfo, it) }
```

with `checkUpdate`, right?

Reply:
Missed this.